fastly/resource_fastly_service_dictionary_items_v1_test.go

```diff
@@ -0,0 +1,483 @@
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceDictionaryItemsV1RemoteState(&service, name, dictName, expectedRemoteItems),
+					resource.TestCheckResourceAttr("fastly_service_dictionary_items_v1.items", "items.%", "2"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccFastlyServiceDictionaryItemV1_create_dynamic(t *testing.T) {
+	var service gofastly.ServiceDetail
+	name := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+	dictName := fmt.Sprintf("dict %s", acctest.RandString(10))
+
+	expectedRemoteItems := map[string]string{
+		"alpha": "alpha.demo.notexample.com",
+		"beta":  "beta.demo.notexample.com",
+		"gamma": "gamma.demo.notexample.com",
+		"delta": "delta.demo.notexample.com",
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckServiceV1Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceDictionaryItemsV1Config_create_dynamic(name, dictName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.myservice", &service),
+					testAccCheckFastlyServiceDictionaryItemsV1RemoteState(&service, name, dictName, expectedRemoteItems),
+					resource.TestCheckResourceAttr("fastly_service_dictionary_items_v1.common", "items.%", "4"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccFastlyServiceDictionaryItemV1_update(t *testing.T) {
+	var service gofastly.ServiceDetail
+	name := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+	dictName := fmt.Sprintf("dict %s", acctest.RandString(10))
+
+	expectedRemoteItems := map[string]string{
+		"key1": "value1",
+		"key2": "value2",
+	}
+
+	expectedRemoteItemsAfterUpdate := map[string]string{
+		"key1": "valueOne",
+		"key2": "value2",
+		"key3": "value3",
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckServiceV1Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceDictionaryItemsV1Config_create(name, dictName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceDictionaryItemsV1RemoteState(&service, name, dictName, expectedRemoteItems),
+					resource.TestCheckResourceAttr("fastly_service_dictionary_items_v1.items", "items.%", "2"),
+				),
+			},
+			{
+				Config: testAccServiceDictionaryItemsV1Config_update(name, dictName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceDictionaryItemsV1RemoteState(&service, name, dictName, expectedRemoteItemsAfterUpdate),
+					resource.TestCheckResourceAttr("fastly_service_dictionary_items_v1.items", "items.%", "3"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccFastlyServiceDictionaryItemV1_external_item_is_ignored(t *testing.T) {
```

Comment:
Maybe `TestAccFastlyServiceDictionaryItemV1_external_item_is_removed` is a better description for the new behaviour

Reply:
The tests have been adjusted to reflect the correct expectations.